### PR TITLE
Sliding Window Eval + Muon6 (val_bpb 1.1973)

### PR DESCRIPTION
**Pull request by beee003**
## Summary

- **Mean val_bpb: 1.1973** (3 seeds: 1337→1.1968, 42→1.1974, 7→1.1978, p<0.001)
- Artifact: ~15.9 MB (under 16 MB)
- Training: ~13,688 steps in 600s on 8xH100 SXM
- Eval: ~126s sliding window (within 10 min budget)

## Key Techniques

1. **Sliding Window Evaluation (stride=256)**: Each token scored with 768+ context instead of 0-1023 average. Added `forward_logits()` method for efficient inference.
2. **Muon 6-step Newton-Schulz**: More accurate gradient orthogonalization (`MUON_BACKEND_STEPS=6`).
3. **Extended Momentum Warmup**: `MUON_MOMENTUM_WARMUP_STEPS=1000` (up from 500) stabilizes early training.
4. **Longer Warmdown**: `WARMDOWN_ITERS=1500` (up from 1200) for smoother LR decay.

## Reproduction

```bash
MUON_BACKEND_STEPS=6 MUON_MOMENTUM_WARMUP_STEPS=1000 WARMDOWN_ITERS=1500 EVAL_STRIDE=256 \
SEED=1337 RUN_ID=submission VAL_LOSS_EVERY=0 \
torchrun --standalone --nproc_per_node=8 train_gpt.py
```

## Results

| Seed | val_loss | val_bpb | Steps |
|------|----------|---------|-------|
| 1337 | 2.0208 | 1.1968 | 13,688 |
| 42 | 2.0217 | 1.1974 | 13,688 |
| 7 | 2.0225 | 1.1978 | 13,688 |
| **Mean** | **2.0217** | **1.1973** | |

### CODE REVIEW

**Comment by MatoTeziTanka** [comment]
### Community Review — Sliding Window Eval + Muon6 (val_bpb 1.1973)

**BPB:** 1.1973 | **Compliance:** LOOKS CLEAN — pure-neural submission, no TTT/SLOT/n-gram-cache

**What I found in the code** (head SHA `afdc3d5693b5`, file `records/track_10min_16mb/2026-03-20_SlidingWindowMuon6/train_gpt.py`):

Static code review found no TTT adaptation function, no SLOT optimization loop, no n-gram-cache class, and no pre-quant val-token fine-tune. The eval path uses the standard sliding-window stride-64 pattern. The submission is a pure-neural architecture iteration on the standard SP1024/SP4096/SP8192 baseline.

**CPU smoke test (CT2038 proteus-engine, 2026-04-11):** import OK in 0.03s, dim=512, layers=9, vocab=1024, code=55272 B, SMOKE_TEST_PASS

**Verdict:** LOOKS CLEAN.

**Recommendation to @cocohearts @valerio-oai @0hq @yuzhougu-oai @notapplica:** **MERGE** pending the usual record-track checks (3-seed validation, under-16MB artifact cap, ≤600s train + ≤600s eval on 8×H100 SXM). No compliance flags from the classification pass — this looks like a clean pure-neural iteration on the standard baseline.

*Auto-classification caveat:* this review was drafted by the AST-based classifier. If there's a non-standard eval mechanism (logit postprocessing, hedge mixing, etc.) that I missed because it's factored into a helper file or a non-standard function name, please flag it and I'll re-run the audit manually.

---
*Reviewed by [@MatoTeziTanka](https://github.com/MatoTeziTanka) — [The Agora](https://matotezitanka.github.io/parameter-golf). CPU smoke test (CT2038 proteus-engine, 2026-04-11): import OK in 0.03s, dim=512, layers=9, vocab=1024, code=55272 B, SMOKE_TEST_PASS. Classification via deterministic AST-based `classify_prs.py` (pattern bank derived from ~65 manually-reviewed PRs earlier in the 2026-04-11 sweep). This review was auto-drafted from a template and spot-checked before posting — if the template misread your code, please call it out so I can iterate the classifier.*
